src/Mod/Sketcher/Gui/TaskSketcherElements.cpp

```diff
@@ -295,54 +383,141 @@ TaskSketcherElements::TaskSketcherElements(ViewProviderSketch *sketchView)
         this                     , SLOT  (on_listWidgetElements_filterShortcutPressed())
        );
     QObject::connect(
-        ui->comboBoxElementFilter, SIGNAL(currentIndexChanged(int)),
-        this                     , SLOT  (on_listWidgetElements_currentFilterChanged(int))
-       );
+        ui->listMultiFilter, SIGNAL(itemChanged(QListWidgetItem*)),
+        this, SLOT(on_listMultiFilter_itemChanged(QListWidgetItem*))
+    );
     QObject::connect(
-        ui->comboBoxModeFilter, SIGNAL(currentIndexChanged(int)),
-        this                     , SLOT  (on_listWidgetElements_currentModeFilterChanged(int))
+        ui->filterBox, SIGNAL(stateChanged(int)),
+        this, SLOT(on_filterBox_stateChanged(int))
     );
     QObject::connect(
-        ui->namingBox, SIGNAL(stateChanged(int)),
-        this                     , SLOT  (on_namingBox_stateChanged(int))
-       );
+        ui->settingsButton, SIGNAL(clicked(bool)),
+        this, SLOT(on_settingsButton_clicked(bool))
+    );
     QObject::connect(
-        ui->autoSwitchBox, SIGNAL(stateChanged(int)),
-        this                     , SLOT  (on_autoSwitchBox_stateChanged(int))
-       );
+        qAsConst(ui->settingsButton)->actions()[0], SIGNAL(changed()),
+        this, SLOT(on_settings_extendedInformation_changed())
+    );
 
     connectionElementsChanged = sketchView->signalElementsChanged.connect(
         boost::bind(&SketcherGui::TaskSketcherElements::slotElementsChanged, this));
 
     this->groupLayout()->addWidget(proxy);
 
-    ui->comboBoxElementFilter->setCurrentIndex(0);
-    ui->comboBoxModeFilter->setCurrentIndex(0);
 
-    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Sketcher/Elements");
+    slotElementsChanged();
 
-    ui->autoSwitchBox->setChecked(hGrp->GetBool("Auto-switch to edge", true));
-    ui->namingBox->setChecked(hGrp->GetBool("Extended Naming", false));
+    // make filter items checkable
+    ui->listMultiFilter->blockSignals(true);

```

Comment:
Don't do that. You shall store return variable 
```suggestion
    sigblk = ui->listMultiFilter->blockSignals(true);
```
then when you unblock restore the value
```
    ui->listMultiFilter->blockSignals(sigblk);
```

This said, easiest solution is to use a scoped QSignalBlocker -- better if an exception comes during the code execution --.

Reply:
Ok. This code was coming from the constraint multi filter window. 
I changed to 

    bool sigblk = ui->listMultiFilter->blockSignals(true);
    ...
    ui->listMultiFilter->blockSignals(sigblk);

Reply:
That's it. So as said, a still better way is
```
// code not protected from signal emitting
{
    QSignalBlocker sigblk(ui->listMultiFilter);
    // code to be protected
}
// signal no more blocked
```